support-workers/src/typescript/model/address.ts

```diff
@@ -0,0 +1,108 @@
+import { z } from 'zod';
+
+export const addressSchema = z.object({
+	lineOne: z.string().nullable(),
+	lineTwo: z.string().nullable(),
+	city: z.string().nullable(),
+	state: z.string().nullable(),
+	postCode: z.string().nullable(),
+	country: z.string(), //TODO: build a schema for this
+});
+
+type AddressLine = {
+	streetNumber?: string;
+	streetName: string;
+};
+
+export function combinedAddressLine(
+	addressLine1: string | null,
+	addressLine2: string | null,
+): AddressLine | undefined {
+	const singleAddressLine = (addressLine: string): AddressLine => {
+		const pattern = /([0-9]+) (.+)/;
+		const match = addressLine.match(pattern);
+		if (match) {
+			const [, streetNumber, streetName] = match;
+			return { streetNumber, streetName: streetName ?? '' };
+		} else {
+			return { streetName: addressLine };
+		}
+	};
+
+	const addressLine1MaybeSplit = addressLine1
+		? singleAddressLine(addressLine1)
+		: undefined;
+	const addressLine2MaybeSplit = addressLine2
+		? singleAddressLine(addressLine2)
+		: undefined;
+
+	const concatStreetNames = (
+		firstStreetName: string,
+		secondStreetName: string,
+	): string => `${firstStreetName}, ${secondStreetName}`;
+
+	const combinedLine = (() => {
+		if (!addressLine1MaybeSplit && !addressLine2MaybeSplit) {
+			return undefined;
+		}
+		if (addressLine1MaybeSplit && !addressLine2MaybeSplit) {
+			return addressLine1MaybeSplit;
+		}
+		if (!addressLine1MaybeSplit && addressLine2MaybeSplit) {
+			return addressLine2MaybeSplit;
+		}
+		if (addressLine1MaybeSplit && addressLine2MaybeSplit) {
+			if (addressLine1MaybeSplit.streetNumber) {
+				return {
+					streetNumber: addressLine1MaybeSplit.streetNumber,
+					streetName: concatStreetNames(
+						addressLine1MaybeSplit.streetName,
+						addressLine2MaybeSplit.streetName,
+					),
+				};
+			} else if (addressLine2MaybeSplit.streetNumber) {
+				return {
+					streetNumber: addressLine2MaybeSplit.streetNumber,
+					streetName: concatStreetNames(
+						addressLine2MaybeSplit.streetName,
+						addressLine1MaybeSplit.streetName,
+					),
+				};
+			} else {
+				return {
+					streetName: concatStreetNames(
+						addressLine1MaybeSplit.streetName,
+						addressLine2MaybeSplit.streetName,
+					),
+				};
+			}
+		}
+		return undefined;
+	})();
+
+	return combinedLine;
+}
+
+export function clipForZuoraStreetNameLimit(
+	addressLine: AddressLine,
+): AddressLine {
+	if (addressLine.streetName.length > 100) {
+		return { ...addressLine, streetName: addressLine.streetName.slice(0, 100) };
+	}
+	return addressLine;
```

Comment:
might simplify it if we just limit it regardless?
```suggestion
	return { ...addressLine, streetName: addressLine.streetName.slice(0, 100) };
```